library/ssl_cli.c

```diff
@@ -2244,23 +2254,7 @@ static int ssl_parse_server_hello( mbedtls_ssl_context *ssl )
      */
     comp = buf[37 + n];
 
-    int bad_comp = 0;
-#if defined(MBEDTLS_ZLIB_SUPPORT)
-    /* See comments in ssl_write_client_hello() */
-    accept_comp = 1;
-#if defined(MBEDTLS_SSL_PROTO_DTLS)
-    if( ssl->conf->transport == MBEDTLS_SSL_TRANSPORT_DATAGRAM )
-        accept_comp = 0;
-#endif
-
-    if( comp != MBEDTLS_SSL_COMPRESS_NULL &&
```

Comment:
I think that the problem lies here: previously, `ssl->conf->transport == MBEDTLS_SSL_TRANSPORT_DATAGRAM` was only a problem if there was any compression other than `MBEDTLS_SSL_COMPRESS_NULL` used, and now it will always error out with ZLIB defined.

Reply:
You can divide it now to
```
if( ssl->conf->transport == MBEDTLS_SSL_TRANSPORT_DATAGRAM &&
    comp != MBEDTLS_SSL_COMPRESS_NULL )
    bad_comp = 1;
    
if( comp != MBEDTLS_SSL_COMPRESS_NULL &&
    comp != MBEDTLS_SSL_COMPRESS_DEFLATE )
    bad_comp = 1; 
```
this way it's clear what's the purpose of each of these checks.

Reply:
Ah, I misread the parentheses when I made the transformation - thanks!